### PR TITLE
Add MDNS and LLMNR status to API

### PR DESCRIPTION
**Pull request by mdegat01**
## Proposed change

Add MDNS and LLMNR status to `/dns/info`. Returns true if MDNS and LLMNR are enabled in systemd-resolved, returns false if they are disabled or systemd-resolved is unreachable (as then the DNS plugin will not be able to resolve these names).

Also added `llmnr_hostname` to `/host/info`. Usually this will be identical to `hostname`. But it is possible for it to be different if there was a conflict on the network (like say if there was another machine already named `homeassistant`). Figured it might help someone debug a tricky situation.

Lastly added `resolved` to host features in `/host/info`. I missed this when I added resolved, I feel like it should be there.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast supervisor tests`)
- [x] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Comment by mdegat01** [comment]
@pvizeli I should note that when going through this I realized something. I cannot add `mdns` and `llmnr` to `/dns/options` currently because the DBus API of `org.freedesktop.resolve1` doesn't allow you to change the global default for mdns and llmnr, which is what we're returning here. It only allows you to change the mdns and llmnr setting per network interface.

Which means it is possible for this to return `true` but llmnr and/or mdns still not to work. I imagine this is pretty unlikely but technically possible.

Do you think we need to account for this possibility? If so maybe I can move these attributes to `/host/info` to reflect that they're global settings and then map the info from the current active interface into `/dns/info`. I would like to do that in a separate PR though since I'd have to explore [org.freedesktop.resolve1.Link](https://www.freedesktop.org/software/systemd/man/org.freedesktop.resolve1.html#Link%20Object) and see how the info in NetworkManager matches up (since org.freedesktop.resolve1 just has a `GetLink` method, not a `ListLinks` method).

**Comment by agners** [comment]
> Which means it is possible for this to return true but llmnr and/or mdns still not to work because it has been turned off for the active interface. I imagine this is pretty unlikely but technically possible. Do you think we need to account for this possibility? If so maybe I can move these attributes to /host/info to reflect that they're global settings and then map the info from the current active interface into /dns/info.

While it is nice, I think I'd rather prefer to have it not at all then it to be "unreliable". IMHO, it should reflect the current active network profiles setting and be in the dns options.

Exposing the global setting in `/host/info`, doesn't seem to be worth it.

File: supervisor/plugins/dns.py [hunk 1]
```diff
@@ -98,6 +100,22 @@ def latest_version(self) -> Optional[AwesomeVersion]:
         """Return latest version of CoreDNS."""
         return self.sys_updater.version_dns
 
+    @property
+    def mdns(self) -> bool:
+        """MDNS hostnames can be resolved."""
+        return self.sys_dbus.resolved.multicast_dns in [
+            MulticastProtocolEnabled.YES,
+            MulticastProtocolEnabled.RESOLVE,
```
Comment (pvizeli) [inline supervisor/plugins/dns.py:108]: same?

File: supervisor/plugins/dns.py [hunk 2]
```diff
@@ -98,6 +100,22 @@ def latest_version(self) -> Optional[AwesomeVersion]:
...
+        ]
+
+    @property
+    def llmnr(self) -> bool:
+        """LLMNR hostnames can be resolved."""
+        return self.sys_dbus.resolved.llmnr in [
+            MulticastProtocolEnabled.YES,
+            MulticastProtocolEnabled.RESOLVE,
```
[the remark this answers is not in the document]
Comment (mdegat01) [reply supervisor/plugins/dns.py:116]: As discussed, left this alone and instead added `broadcast_mdns` and `broadcast_llmnr` fields to `/host/info` so users could debug why they weren't broadcasting expected names as well.

**Review by pvizeli — APPROVED** [review]
Let's start with that and looking how it goes

**Comment by ludeeus** [comment]
Remember to create a dev documentation PR when the API changes 👍

**Review by agners — APPROVED** [review]
LGTM.

**Pull request merged**